src/Microsoft.Azure.WebJobs.Host/Bindings/OpenType.cs

```diff
@@ -325,6 +325,10 @@ public override bool IsMatch(Type type, OpenTypeMatchContext context)
                 {
                     return false;
                 }
+                if (type.GetInterface("System.IDisposable") != null)
```

Comment:
The key thing here is that POCOs shouldn't implement IDisposable. Poco is a very broad rule, and our "bind to poco" is hiding user errors because it successfully binds to too much. So we're trying to tame it and exclude non-pocos. Pocos should not implement IDisposable.